src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -0,0 +1,575 @@
+         *        none
+         * @param hoverAction The hover action of the text, or {@code null} for
+         *        none
+         * @param shiftClickAction The shift click action of the text, or
+         *        {@code null} for none
+         * @param translation The translation of the text
+         * @param arguments The arguments for the translation
+         */
+        public Translatable(TextColor color, TextStyle style, ImmutableList<Text> children, @Nullable ClickAction<?> clickAction,
+                @Nullable HoverAction<?> hoverAction, @Nullable ShiftClickAction<?> shiftClickAction, Translation translation,
+                ImmutableList<Object> arguments) {
+            super(color, style, children, clickAction, hoverAction, shiftClickAction);
+            this.translation = checkNotNull(translation, "translation");
+            this.arguments = checkNotNull(arguments, "arguments");
+        }
+
+        /**
+         * Returns the translation of this {@link Text}.
+         *
+         * @return The translation of this text
+         */
+        public final Translation getTranslation() {
+            return this.translation;
+        }
+
+        /**
+         * Returns the list of {@link Translation} arguments used to format this
+         * {@link Text}.
+         *
+         * @return The list of translation arguments
+         */
+        public final ImmutableList<Object> getArguments() {
+            return this.arguments;
+        }
+
+        @Override
+        public TextBuilder.Translatable builder() {
+            return new TextBuilder.Translatable(this);
+        }
+
+        @Override
+        public boolean equals(@Nullable Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof Translatable) || !super.equals(o)) {
+                return false;
+            }
+
+            Translatable that = (Translatable) o;
+            return this.translation.equals(that.translation)
+                    && this.arguments.equals(that.arguments);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(super.hashCode(), this.translation, this.arguments);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .addValue(super.toString())
+                    .add("translation", this.translation)
+                    .add("arguments", this.arguments)
+                    .toString();
+        }
+
+    }
+
+    /**
+     * Represents a {@link Text} containing a selector that will be replaced by
+     * the names of the matching entities on the client.
+     *
+     * @see org.spongepowered.api.text.selector.Selector
+     * @see TextBuilder.Score
+     */
+    public static class Selector extends Text {
+
+        protected final org.spongepowered.api.text.selector.Selector selector;
```

Comment:
Having to qualify this fully is a shame, but I can't think of any better way short of renaming this class to something like `Select` (which may be a good idea because convenience)